dashboard/app/controllers/pd/workshop_enrollment_controller.rb

```diff
@@ -34,17 +34,15 @@ def new
           workshop_enrollment_status: "full"
         }.to_json
       }
-    elsif (csd_or_csp_workshop || @workshop.subject == Pd::Workshop::SUBJECT_CSF_201) && !current_user
+    elsif !current_user
       render :logged_out
-    elsif csd_or_csp_workshop && current_user && current_user.teacher? && !current_user.email.present?
+    elsif current_user.teacher? && !current_user.email.present?
       render '/pd/application/teacher_application/no_teacher_email'
     else
       @enrollment = ::Pd::Enrollment.new workshop: @workshop
```

Comment:
Seems like we will get to this code path if the previous `elsif` returns false, i.e either when `current_user.teacher == false` or `current_user.email.present? == true` or both. That means a student with this link can also request this action. Is that a scenario we want?

Reply:
it's not a scenario that we explicitly support, but it works that way today. I can add another view like the `:logged_out` view explaining that you must be a teacher in order to enroll

Reply:
actually - it looks like we have some [code](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/pd/workshop_enrollment_controller.rb#L189) to upgrade your account automatically if you do register as a student. I'm not familiar with that code and it's outside the scope of this task, so I'm not going to take this on in this pr.

Reply:
I guess it doesn't change the status quo, so no regression to fix.

In a kind-of-related topic, I'm always cautious when I see something like `if condition_a && condition_b ... else ...` because it means 3/4 of possible combinations will go the the `else` path and the author might not intend that. It becomes worse in `if condition_a && condition_b && condition_c ... else ...`. In 7/8 combinations going `else` route, it seems likely to have a surprise.

Reply:
I think that hypothetical is okay if we only want to send 1/8 of situations to a special place, and we expect 7/8 of situations to end up going to the default :)

Reply:
Agree, it's all good when the author is aware of that and confident that there is no corner cases or hidden surprises.